### PR TITLE
[Trello 124] Fix a post processing process

### PR DESCRIPTION
**Pull request by omelniz**

### CODE REVIEW

File: src/store/ducks/posts/saga/postsCreate.js [hunk 1]
```diff
@@ -161,6 +163,65 @@ function* handleTextOnlyPost(req) {
   }
 }
 
+/**
+ *
+ */
+function* handlePostsCreateSuccess(post) {
+  const userId = path(['postedBy', 'userId'], post)
+
```
Comment (omelniz) [inline src/store/ducks/posts/saga/postsCreate.js:171]: Here is a good place to put logic to handle an avatar upload

File: src/store/ducks/posts/saga/postsCreate.js [hunk 2]
```diff
@@ -161,6 +163,65 @@ function* handleTextOnlyPost(req) {
...
+    return post
+  }
+  
+  try {
+    const { idle } = yield race({
+      idle: take(constants.POSTS_CREATE_IDLE),
+      timeout: delay(TIMEOUT_DELAY),
+      completed: take(constants.POSTS_CREATE_COMPLETED),
+    })
+
```
Comment (omelniz) [inline src/store/ducks/posts/saga/postsCreate.js:203]: On idle we need to delete a post

Comment (omelniz) [inline src/store/ducks/posts/saga/postsCreate.js:203]: on complete we need to compare processingPost.postId and completed.postId if they are different run spawn(checkPostsCreateProcessing) again

It gives as an opportunities for parallel uploads

File: src/store/ducks/posts/saga/postsCreate.js (hunk 1 above)
Comment (omelniz) [reply src/store/ducks/posts/saga/postsCreate.js:171]: especially if we add a flag 'isAvatar'

File: src/store/ducks/posts/saga/postsCreate.js (hunk 2 above)
Comment (omelniz) [reply src/store/ducks/posts/saga/postsCreate.js:203]: we can use normalizer to create and manage a queue of uploading posts

File: src/store/ducks/subscriptions/saga.js [hunk 3]
```diff
@@ -338,24 +328,11 @@ function* subscriptionNotificationStart(req) {
...
-     * Fires when one of the user's posts reaches ERROR state
+     * 1. Fires when one of the user's posts reaches COMPLETED state for the first time 
+     * 2. Fires when one of the user's posts reaches ERROR state
      */
-    if (type === 'POST_ERROR') {
-      const data = yield queryService.apiRequest(postsQueries.getPost, { postId })
-      const selector = path(['data', 'post'])
-      yield put(postsActions.postsCreateFailure({ data: {}, payload: selector(data), meta: {} }))
+    if (type === 'POST_COMPLETED' || type === 'POST_ERROR') {
+      yield put(postsActions.postsCreateCompleted())
```
Comment (omelniz) [inline src/store/ducks/subscriptions/saga.js:335]: put here an eventData for compare post ids

File: src/store/ducks/posts/saga/postsCreate.js [hunk 4]
```diff
@@ -161,6 +163,65 @@ function* handleTextOnlyPost(req) {
...
+  try {
+    const { idle } = yield race({
+      idle: take(constants.POSTS_CREATE_IDLE),
+      timeout: delay(TIMEOUT_DELAY),
+      completed: take(constants.POSTS_CREATE_COMPLETED),
+    })
+  
+    if (!idle) {
+      const { post } = yield race({
+        post: retry(3, TIMEOUT_DELAY, checkRequest),
```
Comment (omelniz) [inline src/store/ducks/posts/saga/postsCreate.js:206]: here we can increase the delay every time. 3, 5, 10 seconds. For this needs to rewrite it as a generator function

File: src/store/ducks/subscriptions/saga.js [hunk 5]
```diff
@@ -338,24 +328,11 @@ function* subscriptionNotificationStart(req) {
...
-    /**
-     * Fires when one of the user's posts reaches ERROR state
+     * 1. Fires when one of the user's posts reaches COMPLETED state for the first time 
+     * 2. Fires when one of the user's posts reaches ERROR state
      */
-    if (type === 'POST_ERROR') {
-      const data = yield queryService.apiRequest(postsQueries.getPost, { postId })
-      const selector = path(['data', 'post'])
-      yield put(postsActions.postsCreateFailure({ data: {}, payload: selector(data), meta: {} }))
+    if (type === 'POST_COMPLETED' || type === 'POST_ERROR') {
```
Comment (omelniz) [inline src/store/ducks/subscriptions/saga.js:334]: it also can be fired onForeground

**Comment by azimgd** [comment]
Very nice job!

**Pull request merged**